clientconn.go

```diff
@@ -1281,7 +1280,7 @@ func (ac *addrConn) startHealthCheck(ctx context.Context, newTr transport.Client
 	go func() {
```

Comment:
Remove this "go" since this now should block.

Reply:
It does not necessarily need to block, but it saves creating a new go routine and potential delay due to that, so removed.